app/screens/edit_profile/edit_profile.tsx

```diff
@@ -0,0 +1,308 @@
+            setCanSave(value);
+        } else {
+            const buttons = {
+                rightButtons: [{...rightButton, enabled: value}],
+            };
+
+            setButtons(componentId, buttons);
+        }
+    };
+
+    const submitUser = async () => {
+        enableSaveButton(false);
+        setError(undefined);
+        setUpdating(true);
+        try {
+            const partialUser: Partial<UserProfile> = {
+                email: userInfo.email,
+                first_name: userInfo.firstName,
+                last_name: userInfo.lastName,
+                nickname: userInfo.nickname,
+                position: userInfo.position,
+                username: userInfo.username,
+            };
+
+            const {error: reqError} = await updateMe(serverUrl, partialUser);
+
+            if (reqError) {
+                return resetScreen(reqError as Error);
+            }
+
+            close();
+        } catch (e) {
+            return resetScreen(e as Error);
+        }
+
+        return null;
+    };
+
+    const resetScreen = (resetError: Error) => {
+        setError(resetError?.message);
+        setUpdating(false);
+        enableSaveButton(true);
+        scrollViewRef.current?.scrollToPosition(0, 0, true);
+    };
+
+    const updateField = useCallback((id: string, name: string) => {
+        const update = {...userInfo};
+        update[id] = name;
+        setUserInfo(update);
+
+        // @ts-expect-error access object property by string key
+        const currentValue = currentUser[id];
+        const didChange = currentValue !== name;
+        enableSaveButton(didChange);
+    }, [userInfo]);
+
+    const setScrollViewRef = useCallback((ref) => {
+        scrollViewRef.current = ref;
+    }, []);
+
+    const renderProfilePicture = () => {
+        return (
+            <View style={style.top}>
+                <ProfilePicture
+                    author={currentUser}
+                    size={153}
+                    showStatus={false}
+                />
+            </View>
+        );
+    };
+
+    if (updating) {
+        return <ProfileUpdating/>;
+    }
+
+    return (
+        <>
+            {isTablet &&
+                <TabletTitle
+                    action={rightButton.text}
+                    enabled={canSave}
+                    onPress={submitUser}
+                    testID='custom_status.done.button'
+                    title={intl.formatMessage({id: 'mobile.screen.your_profile', defaultMessage: 'Your Profile'})}
+                />
+            }
+            <SafeAreaView
+                edges={edges}
+                style={style.flex}
+                testID='edit_profile.screen'
+            >
+                <KeyboardAwareScrollView
+                    bounces={false}
+                    innerRef={setScrollViewRef}
+                    testID='edit_profile.scroll_view'
+                >
+                    {error && <ProfileError error={error}/>}
```

Comment:
```suggestion
                    {Boolean(error) && <ProfileError error={error}/>}
```

to avoid react from complaining that you are trying to render a string without a `<Text>` component